app/src/main/java/com/gruutnetworks/gruutsigner/util/AuthCertUtil.java

```diff
@@ -193,13 +197,107 @@ public String getCert(SecurityConstants.Alias alias)
+                                      String x, String y, String time)
+            throws IOException, NoSuchAlgorithmException, CertificateException, NoSuchProviderException,
+            InvalidKeyException, SignatureException {
+
+        byte[] sigTime = ByteBuffer.allocate(8).putLong(Integer.parseInt(time)).array();
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        outputStream.write(Base64.decode(mNonce, Base64.NO_WRAP));  // 256 bit
+        outputStream.write(Base64.decode(sNonce, Base64.NO_WRAP));  // 256 bit
+        outputStream.write(Hex.decode(x));                          // 256 bit
+        outputStream.write(Hex.decode(y));                          // 256 bit
+        outputStream.write(sigTime);                                // 64 bit
+
+        byte[] mergerSig = outputStream.toByteArray();
+        outputStream.close();
+
+        return authCertUtil.verifyData(mergerSig, signature, cert);
+    }
+
+    /**
+     * MSG_REQ_SSIG에 대한 지지서명 생성
+     *
+     * @param sender    signer의 id. int value...?
+     * @param time      timestamp
+     * @param mId       요청한 merger의 id
+     * @param cId       local chain id
+     * @param hgt       block height
+     * @param tx        transaction root
+     * @return support signature string
+     */
+    public String generateSupportSignature(String sender, String time, String mId, String cId, String hgt, String tx)
+            throws IOException, CertificateException, InvalidKeyException, NoSuchAlgorithmException,
+            KeyStoreException, SignatureException, UnrecoverableEntryException {
+
+        byte[] sigSender = ByteBuffer.allocate(8).putLong(Integer.parseInt(sender)).array();
+        byte[] sigTime = ByteBuffer.allocate(8).putLong(Integer.parseInt(time)).array();
+        byte[] sigHgt = ByteBuffer.allocate(8).putLong(Integer.parseInt(hgt)).array();
+
+        ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+        outputStream.write(sigSender);                          // 64 bit
+        outputStream.write(sigTime);                            // 64 bit
+        outputStream.write(Base64.decode(mId, Base64.NO_WRAP)); // 64 bit
+        outputStream.write(Base64.decode(cId, Base64.NO_WRAP)); // 64 bit
+        outputStream.write(sigHgt);                             // 64 bit
+        outputStream.write(Base64.decode(tx, Base64.NO_WRAP));  // 256 bit
+
+        String signature = authCertUtil.signData(outputStream.toByteArray());
+        outputStream.close();
+
+        return signature;
+    }
+
     /**
      * Signs the data using the key pair stored in the Android Key Store.
      * This signature can be used with the data later to verify it was signed by this application.
      *
      * @return A string encoding of the data signature generated
      */
-    public String signData(byte[] data)
+    private String signData(byte[] data)
```

Comment:
```suggestion
    private String sign(byte[] data)
```
내 관점에서 만드는 서명은 내 꺼니까. sign으로 충분하지 않을까요?